### DISCUSSION TITLE
Allow order specific address (don't store address in address book)

### DISCUSSION THREAD

**Discussion by piotrpog** [craftcms/commerce#2457] [Ideas]
I am lf course talking about situation when user is logged in.

Would that be possible?

**Comment by pdaleramirez** [comment]
@piotrpog What is your use case? cause new address is always created to relate it to an order.

**Comment by piotrpog** [comment]
@pdaleramirez 
Client asked to have functionality like that - new addresses added only from "my addresses" page. During order finalization process, user would choose from precreated address, which would fill address form, and then this address could be further tweaked using form. But tweaked address should not be saved.

**Comment by lukeholder** [comment]
Thanks for the request. We have added it as an enhancement request.

In the mean-time you will need to figure out another way. 

On order complete, you could loop over the customers address book and remove the addresses that are the same as the one attached to the order.

Something like:

```php
Event::on(
    Order::class,
    Order::EVENT_AFTER_COMPLETE_ORDER,
    function(Event $event) {
        // @var Order $order
        $order = $event->sender;
        foreach (Plugin::getInstance()->getAddresses()->getAddressesByCustomerId($order->customerId) as $address)
        {
            if ($address->sameAs($order->getShippingAddress()))
            {
                Plugin::getInstance()->getAddresses()->deleteAddressById($address->id);
            }
        }
    }
);
```

**Comment by piotrpog** [comment]
@lukeholder 
Thank you!
